projects/media-viewer/src/lib/store/reducers/bookmarks.reducer.ts

```diff
@@ -19,47 +20,61 @@ export function bookmarksReducer (state = initialBookmarksState,
       return {
         ...state,
         loading: true
-      }
+      };
     }
 
     case fromBookmarks.LOAD_BOOKMARKS_SUCCESS:
-    case fromBookmarks.LOAD_BOOKMARKS_FAIL:{
+    case fromBookmarks.LOAD_BOOKMARKS_FAIL: {
       const bookmarks = action.payload.status === 200 ? action.payload.body : [];
       const bookmarkEntities = StoreUtils.generateBookmarkEntities(bookmarks);
+      const bookmarkPageEntities = StoreUtils.groupByKeyEntities(bookmarks, 'pageNumber');
       return {
         ...state,
         bookmarks,
         bookmarkEntities,
+        bookmarkPageEntities,
         loaded: true
-      }
+      };
     }
 
     case fromBookmarks.CREATE_BOOKMARK_SUCCESS: {
       const bookmark: Bookmark = action.payload;
       const bookmarkEntities = {
         ...state.bookmarkEntities,
         [bookmark.id]: bookmark,
-      }
+      };
+      const bookmarkArray = Object.keys(bookmarkEntities).map(key => bookmarkEntities[key]);
```

Comment:
we should already have the bookmarks array (bookmarks), that's what the backend returns

Reply:
I'm not sure what you mean? I think the only reason I'm loading them here is because they are needed in the new bookmarkPageEntities collection...